AdyenActions/Components/Voucher/VoucherComponent.swift

```diff
@@ -57,13 +72,16 @@ public final class VoucherComponent: AnyVoucherActionHandler {
     }
 
     /// :nodoc:
-    private let componentName = "voucher"
+    public func didCancel() {
+        passProvider = nil
+    }
 
     /// Handles await action.
     ///
     /// - Parameter action: The await action object.
     public func handle(_ action: VoucherAction) {
         Analytics.sendEvent(component: componentName, flavor: _isDropIn ? .dropin : .components, context: apiContext)
+        fetchAndCacheAppleWalletPassIfNeeded(with: action)
 
         var viewControllerProvider = voucherViewControllerProvider
             ?? VoucherViewControllerProvider(style: style, environment: apiContext.environment)
```

Comment:
Maybe we can do here same thing as in `AppleWalletPassProvider` ?

```swift
    public init(apiContext: APIContext, style: VoucherComponentStyle?) {
        self.apiContext = apiContext
        self.style = style ?? VoucherComponentStyle()
        let viewControllerProvider = VoucherViewControllerProvider(style: style, environment: apiContext.environment)
        self.init(apiContext: apiContext, voucherViewControllerProvider: viewControllerProvider)
    }
    
    internal convenience init(apiContext: APIContext,
                              voucherViewControllerProvider: AnyVoucherViewControllerProvider) {
        self.voucherViewControllerProvider = voucherViewControllerProvider
        self.apiContext = apiContext
    }
    
```

Default public `init` contains  creates local `voucherViewControllerProvider`

Reply:
do what?

Reply:
sorry, I have edited..
Idea is to resolve dependencies in init and avoid checking "on run": one public `init` that used in code and one internal `init` for injecting mock in tests

Reply:
no we can't, the provider has to be recreated every time the handle function is called with a new action, to invalidate its cache and recreate a new cache of the apple wallet pass

Reply:
But it is never recreated

Reply:
its recreated here https://github.com/Adyen/adyen-ios/blob/cb0f4ed610c5128ce5629c93050011a5686bad74/AdyenActions/Components/Voucher/VoucherComponent.swift#L104